eos/maths/gegenbauer-polynomial.hh

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ * Copyright (c) 2022 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH
+#define EOS_GUARD_EOS_MATHS_GEGENBAUER_POLYNOMIALS_HH 1
+
+#include <eos/maths/power-of.hh>
+
+#include <vector>
+
+namespace eos
+{
+    class GegenbauerPolynomial
+    {
+        private:
+            const unsigned int order;
+
+            const double alpha; // alpha > -0.5
+
+            std::vector<double> coefficients;
+
+            const unsigned int halforder;
```

Comment:
It would be helpful to add some comments at the top of the class to explain the order and halforder of the polynomial(s).